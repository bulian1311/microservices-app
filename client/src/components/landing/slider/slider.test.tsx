import React from "react";
import { render, RenderResult, fireEvent } from "@testing-library/react";

import Slider from "./slider.component";

let documentBody: RenderResult;
const setState = jest.fn();
const useStateSpy = jest.spyOn(React, "useState");
useStateSpy.mockImplementation(() => [null, setState]);

beforeEach(() => {
  documentBody = render(
    <Slider
      images={[]}
      dataTestIdLeft="test-left"
      dataTestIdRight="test-right"
    />
  );
});

test("Slider component, match snapshot", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
});

test("Slider click left.", () => {
  const btn = documentBody.getByTestId("test-left");

  fireEvent.click(btn);

  expect(setState).toHaveBeenCalledWith(-1);
});

test("Slider test right.", () => {
  const btn = documentBody.getByTestId("test-right");

  fireEvent.click(btn);

  expect(setState).toHaveBeenCalledWith(1);
});
