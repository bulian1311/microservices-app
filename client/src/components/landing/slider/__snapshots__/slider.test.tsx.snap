// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component, match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-dlfnbm laaXGn"
    >
      <div
        class="sc-eCssSg fyeJyK"
      >
        <a
          class="sc-bdfBwQ eIosNS"
          data-testid="test-left"
        >
          <svg
            class="sc-gsTCUz iapKVs"
            viewBox="0 0 20 20"
            width="1"
          >
            <path
              d="M20 8.75H4.7875L11.775 1.7625L10 0L0 10L10 20L11.7625 18.2375L4.7875 11.25H20V8.75Z"
              fill="#555555"
              fill-opacity="0.54"
            />
          </svg>
        </a>
      </div>
      <img
        class="sc-hKgILt ikRUVT"
      />
      <div
        class="sc-jSgupP fqAhML"
      >
        <a
          class="sc-bdfBwQ eIosNS"
          data-testid="test-right"
        >
          <svg
            class="sc-gsTCUz iapKVs"
            viewBox="0 0 20 20"
            width="1"
          >
            <path
              d="M10 0L8.2375 1.7625L15.2125 8.75H0V11.25H15.2125L8.2375 18.2375L10 20L20 10L10 0Z"
              fill="#555555"
              fill-opacity="0.54"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</body>
`;
